test/CompressedRelationsTest.cpp

```diff
@@ -1018,6 +1018,91 @@ TEST(CompressedRelationReader, getResultSizeImpl) {
+    const CompressedBlockMetadata::PermutedTriple& expectedFirst,
+    const CompressedBlockMetadata::PermutedTriple& expectedLast) {
+  using ::testing::Eq;
+  return ::testing::AllOf(AD_FIELD(FLT, firstTriple_, Eq(expectedFirst)),
+                          AD_FIELD(FLT, lastTriple_, Eq(expectedLast)));
+}
+
+}  // namespace
+
+// _____________________________________________________________________________
+TEST(CompressedRelationReader, getFirstAndLastTripleIgnoringGraph) {
+  ad_utility::testing::TestIndexConfig testIndexConfig;
+  testIndexConfig.indexType = qlever::Filetype::NQuad;
+  testIndexConfig.turtleInput =
+      "<a> <a> <a> <g1> . <a> <a> <d> <g2> . <a> <a> <h> <g3> ."
+      "<a> <a> <b> <g1> . <a> <a> <e> <g2> . <a> <a> <i> <g3> ."
+      "<a> <b> <c> <g1> . <a> <b> <f> <g2> . <a> <b> <j> <g3> ."
+      "<a> <b> <d> <g1> . <a> <b> <g> <g2> . <a> <b> <k> <g3> .";
+  auto index = ad_utility::testing::makeTestIndex(
+      "getFirstAndLastTripleIgnoringGraph", std::move(testIndexConfig));
+  auto currentSnapshot = index.deltaTriplesManager().getCurrentSnapshot();
+  auto permutationEnum = Permutation::Enum::SPO;
+  const auto& permutation = index.getImpl().getPermutation(permutationEnum);
+  const auto& locatedTriplesPerBlock =
+      currentSnapshot->getLocatedTriplesForPermutation(permutationEnum);
+
+  auto getId = [&index](std::string_view iri) {
+    return TripleComponent{ad_utility::triple_component::Iri::fromIriref(iri)}
+        .toValueId(index.getVocab(), index.encodedIriManager())
+        .value();
+  };
+  auto a = getId("<a>");
+  auto b = getId("<b>");
+  auto c = getId("<c>");
+  auto i = getId("<i>");
+  auto k = getId("<k>");
+  auto g1 = getId("<g1>");
+  auto g2 = getId("<g2>");
+  auto g3 = getId("<g3>");
+
+  auto getTriples = [&permutation, &currentSnapshot, permutationEnum, &index,
+                     &locatedTriplesPerBlock](std::optional<Id> col0,
+                                              std::optional<Id> col1,
+                                              std::optional<Id> graphId) {
+    ScanSpecification scanSpecification{
+        std::move(col0),
+        std::move(col1),
+        std::nullopt,
+        {},
+        graphId.has_value() ? ScanSpecification::GraphFilter::Whitelist(
+                                  {std::move(graphId).value()})
+                            : ScanSpecification::GraphFilter::All()};
+    CompressedRelationReader::ScanSpecAndBlocks metadataAndBlocks{
+        std::move(scanSpecification),
+        permutation.getAugmentedMetadataForPermutation(*currentSnapshot)};
+    const auto& reader =
+        index.getImpl().getPermutation(permutationEnum).reader();
+    return reader.getFirstAndLastTripleIgnoringGraph(metadataAndBlocks,
+                                                     locatedTriplesPerBlock);
+  };
+  using PT = CompressedBlockMetadata::PermutedTriple;
+  using ::testing::Optional;
+  using ::testing::Pair;
+  for (auto graphId :
```

Comment:
Add a short comment here that we check that the first and last triple don't depend on the graph, but always ignore the specified graph.